homeassistant/helpers/config_validation.py

```diff
@@ -1769,10 +1771,32 @@ def STATE_CONDITION_SCHEMA(value: Any) -> dict[str, Any]:
     )
 )
 
+
+def _backward_compat_trigger_schema(value: Any | None) -> Any:
+    """Backward compatibility for trigger schemas."""
+
+    if not isinstance(value, dict):
```

Comment:
I'm not sure restricting to `dict` is correct, I think we should allow any `Mapping`

Reply:
Isn't Mapping a subtype of dict?

Reply:
No..